workers/unity/Packages/io.improbable.gdk.testutils/MockCommandSender.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.Commands;
+using Unity.Entities;
+
+namespace Improbable.Gdk.TestUtils
+{
+    public class MockCommandSender : ICoreCommandSender
+    {
+        private readonly MockWorld world;
+
+        private long nextRequestId = 1;
+
+        private readonly Dictionary<Type, SortedSet<long>> requestIds = new Dictionary<Type, SortedSet<long>>();
+
+        private readonly Dictionary<long, ICommandRequest> outboundRequests = new Dictionary<long, ICommandRequest>();
+
+        internal MockCommandSender(MockWorld world)
+        {
+            this.world = world;
+        }
+
+        public IEnumerable<(CommandRequestId, TRequest)> GetOutboundCommandRequests<TRequest>() where TRequest : ICommandRequest
+        {
+            return requestIds[typeof(TRequest)].Select(id => (new CommandRequestId(id), (TRequest) outboundRequests[id]));
+        }
+
+        public CommandRequestId SendCommand<TRequest>(TRequest request, Entity sendingEntity = default) where TRequest : ICommandRequest
```

Comment:
It may make sense to break this class into two as currently it does two different things. 

```csharp
public MockCommandSender : ICommandSender
{
	private OutgoingCommands outgoingCommands;
	public CommandRequestId SendCommand<TRequest>(...) { outgoingCommands.Add(...); }
}

public OutgoingCommands 
{
	public void GenerateResponses<TRequest, TResponse(...) {}
	public void GenerateResponse<TRequest, TResponse>(...) {}
}
```

And only giving public access to `OutgoingCommands` in the `MockWorld`. Currently this does two things and a user shouldn't have access to the `ICommandSender` directly as its implementation detail.

Reply:
How about just putting the dictionaries and `GenerateResponse[s]` functions in the `MockConnectionHandler` so that it conforms to the existing API where you call `mockWorld.Connection.(do spatial stuff)` instead of having an additional object in `MockWorld`?

Reply:
Yeah that could work!